prov/tcp/src/tcpx_ep.c

```diff
@@ -188,9 +188,12 @@ static int tcpx_ep_shutdown(struct fid_ep *ep, uint64_t flags)
 static int tcpx_bind_to_port_range(SOCKET sock, void* src_addr, size_t addrlen)
 {
 	int ret, i, rand_port_number;
+	static uint32_t seed;
+	if (!seed)
+		seed = ofi_generate_seed();
 
-	rand_port_number = rand() % (port_range.high + 1 - port_range.low) +
-			   port_range.low;
+	rand_port_number = ofi_xorshift_random(seed) %
```

Comment:
This results in rand_port_number being the same every time through this code path.  Looking at the other use of ofi_xorshift_random, we can change its input parameter to be by reference and update it.  Or we need to update seed here before selecting the port number.

Reply:
Thank you very much for the review. You are absolutely right! I don't get how I missed that. I'm going to update the patch serie.